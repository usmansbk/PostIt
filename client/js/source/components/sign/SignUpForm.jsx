import React from 'react';
import InputField from '../common/InputField.jsx';
import Button from '../common/Button.jsx';

export default class SignUpForm extends React.Component {

  constructor(props) {
    super(props);
    this.validatePassword = this.validatePassword.bind(this);
    this.state = { password: '', confirm: '', isInvalid: false };
  }

  validatePassword(event) {
    const value = event.target.value;
    const id = event.target.id;
    this.setState({
      [id]: value
    }, () => {
      if (this.state.confirm !== this.state.password) {
        this.setState({isInvalid: true});
        return;
      }
      this.setState({isInvalid: false});
    });
  }

  render() {
   const error = <p className='red-text'>Password doesn't match</p>
    return (
      <form>

        <InputField id='username' type='text' required='true' label='Username' className='input-field'>
          Name can only contain letters, numbers and underscore
        </InputField>

        <InputField id='email' type='email' required='true' label='Email address' className='input-field'/>

        <InputField id='password' type='password' required='true' value={this.state.password} label='Password' validate='validate' className='input-field' onChange={this.validatePassword} >
          Password must be 8-32 characters long
        </InputField>

        <InputField id='confirm' type='password' required='true' label='Confirm password' className='input-field' onChange={this.validatePassword} value={this.state.fconfirm} />
        { this.state.isInvalid && error }
        <Button type='submit' className='center-align' color='blue' value='Sign Up'>send</Button>

     </form>
    );
  }
}
