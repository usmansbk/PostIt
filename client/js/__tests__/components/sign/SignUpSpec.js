import React from 'react';
import SignUpForm from '../../../src/components/sign/SignUpForm';
import renderer from 'react-test-renderer';

describe('<SignUpForm />', () => {
	test('renders sign-up form', () => {
		const component = renderer.create(
			<SignUpForm />
		);

		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});