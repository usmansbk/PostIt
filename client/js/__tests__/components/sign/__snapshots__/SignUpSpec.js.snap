// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> renders sign-up form 1`] = `
<form>
  <div
    className="input-field"
  >
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      className="input-field"
      id="username"
      label="Username"
      required="true"
      type="text"
    >
      Name can only contain letters, numbers and underscore
    </input>
    <span
      className="helper-text"
    >
      Name can only contain letters, numbers and underscore
    </span>
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="email"
    >
      Email address
    </label>
    <input
      className="input-field"
      id="email"
      label="Email address"
      required="true"
      type="email"
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="input-field"
      id="password"
      label="Password"
      onChange={[Function]}
      required="true"
      type="password"
      validate="validate"
      value=""
    >
      Password must be 8-32 characters long
    </input>
    <span
      className="helper-text"
    >
      Password must be 8-32 characters long
    </span>
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="confirm"
    >
      Confirm password
    </label>
    <input
      className="input-field"
      id="confirm"
      label="Confirm password"
      onChange={[Function]}
      required="true"
      type="password"
      value={undefined}
    />
  </div>
  <div
    className="center-align"
  >
    <button
      className="btn waves-effect waves-light blue"
      type="submit"
    >
      Sign Up
      <i
        className="material-icons right"
      >
        send
      </i>
    </button>
  </div>
</form>
`;
